catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js

```diff
@@ -140,77 +159,101 @@ function (ich,Marionette,Backbone,ConfigurationEdit,Service,Utils,wreqr,_,$,moda
                 configs.add(currentService);
             }
             if (!_.isUndefined(disabledConfigs)) {
-                disabledConfigs.each(function(config) {
+                disabledConfigs.each(function (config) {
                     configs.add(config.get('service'));
                 });
             }
             return configs;
         },
+        getActions: function (action) {
+            var disabledConfigs = this.model.get('disabledConfigurations');
+            var currentConfig = this.model.get('currentConfiguration');
+            var actions;
+            if (!_.isUndefined(currentConfig)) {
+                actions = currentConfig.get(action);
+            }
+            if (!_.isUndefined(disabledConfigs)) {
+                var actionConfig = _.find(disabledConfigs.models, function (config) {
+                    return config.get(action);
+                });
+                if (actionConfig) {
+                    actions = actionConfig.get(action);
+                }
+            }
+            _.each(actions, function (action) {
+                action.id = action.id.split('.').join('-');
+            });
+            return actions;
+        },
+        handelAction: function (event) {
+            var link = this.$(event.currentTarget);
+            var action = link.attr('action');
+            var actionId = link.attr('action-id');
+            var id = link.attr('id');
+            var failed = $(this.$('#' + id + '-failed')[0]);
+            var success = $(this.$('#' + id + '-success')[0]);
+            var spinner = $(this.$('#' + id + '-spinner')[0]);
+            var httpMethod = this.getHttpMethod(actionId);
+            link.addClass('inactive-link');
+            spinner.show();
+            failed.hide();
+            success.hide();
+            $.ajax({
```

Comment:
Shouldn't be doing ajax actions from a view. Going back and forth to the server is the responsibility of a model in backbone land.